### PR TITLE
Integrate new observer system

### PR DESCRIPTION
**Pull request by basine**
Needs release with https://github.com/wireapp/wire-ios-sync-engine/pull/255

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -23,7 +23,7 @@ github "wireapp/wire-ios-cryptobox" "8.0.0"
 github "wireapp/wire-ios-link-preview" "4.1.0"
 github "wireapp/wire-ios-utilities" "13.1.0"
 github "wireapp/wire-ios-images" "7.1.0"
-github "wireapp/wire-ios-transport" "18.0.2"
+github "wireapp/wire-ios-transport" "18.0.0"
```
Comment (mikeger) [inline Cartfile.resolved]: I guess it's a merge artifact, should not go from 18.0.2 to 18.0.0

File: Wire-iOS/Sources/Analytics/AnalyticsConversationListObserver.m [hunk 2]
```diff
@@ -81,15 +76,12 @@ - (void)setObserving:(BOOL)observing
...
-        self.conversationListObserverToken = [[SessionObjectCache sharedCache].allConversations addConversationListObserver:self];
+        self.conversationListObserverToken = [ConversationListChangeInfo addObserver:self forList:[SessionObjectCache sharedCache].allConversations];
 
         [self performSelector:@selector(probablyReceivedFullConversationList)
                    withObject:nil
                    afterDelay:PermantentConversationListObserverObservationFinalTime];
     }
-    else {
-        [[SessionObjectCache sharedCache].allConversations removeConversationListObserverForToken:self.conversationListObserverToken];
-    }
```
Comment (mikeger) [inline Wire-iOS/Sources/Analytics/AnalyticsConversationListObserver.m]: So setting `observing` = NO would not work any more?

File: Wire-iOS/Sources/Components/AccentColorChangeHandler.m [hunk 3]
```diff
@@ -59,16 +59,15 @@ - (id)initWithObserver:(id)observer handlerBlock:(AccentColorChangeHandlerBlock)
...
     }
     return self;
 }
 
 - (void)dealloc
 {
     self.observer = nil;
-    
-    [ZMUser removeUserObserverForToken:self.userObserverToken];
+    self.userObserverToken = nil;
```
Comment (mikeger) [inline Wire-iOS/Sources/Components/AccentColorChangeHandler.m]: Not necessary, it would be released automatically.

**Comment by mikeger** [comment]
@basine  Uh, we observer a lot! Thanks for porting to the new API. Just a check, did you run the test target? It might rely on the observer method names.

File: Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift [hunk 4]
```diff
@@ -129,6 +129,12 @@ class SettingsTableViewController: SettingsBaseTableViewController {
     override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
         fatalError()
     }
+    
+    deinit {
+        if let token = selfUserObserver {
+            UserChangeInfo.remove(observer: token, forBareUser: nil)
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift]: Isn't this unnecessary?

Comment (basine) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift]: Yeah, obviously overlooked that one.

Comment (basine) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift]: Forgot to remove all removals in Swift :-/

File: Wire-iOS/Sources/Analytics/AnalyticsConversationListObserver.m (hunk 2 above)
Comment (basine) [inline Wire-iOS/Sources/Analytics/AnalyticsConversationListObserver.m]: Ah right, good catch

**Comment by mikeger** [comment]
@basine please merge when it's there

**Comment by mikeger** [comment]
@basine @marcoconti83  I updated the data model, all builds but fails on runtime with error:
```
fatal error: use of unimplemented initializer 'init()' for class 'ZMCDataModel.ContextDidSaveNotificationPersistence'
```

**Comment by daehn** [comment]
@mikeger thats because we did not merge https://github.com/wireapp/wire-ios/pull/659 yet but the data-model has already been released.

**Pull request merged**